stdlib/src/collections/string.mojo

```diff
@@ -1565,25 +1565,34 @@ struct String(
             raise Error("Separator cannot be empty.")
         if str_byte_len < 0:
             output.append("")
+        var ptr = self.unsafe_ptr()
+        alias SelfSlice = StringSlice[__lifetime_of(self)]
+
+        @always_inline("nodebug")
+        fn build_slice(s_ptr: UnsafePointer[UInt8], length: Int) -> SelfSlice:
```

Comment:
NIT: Just for convention, I think it's better to prefix the function with `_` to improve readability, implying that it's not a public function imported from somewhere:
```suggestion
        fn _build_slice(s_ptr: UnsafePointer[UInt8], length: Int) -> SelfSlice:
```

Reply:
It's still defined withing the scope of the function, so I didn't see a need to make it's name "private" since no other func can access it. But I'll change them if you want to :man_shrugging: